src/main/java/io/stargate/sgv2/jsonapi/service/projection/ProjectionLayer.java

```diff
@@ -89,49 +81,51 @@ private static ProjectionLayer buildLayers(
           failOnOverlap,
           DocumentConstants.Fields.DOC_ID,
           root,
-          new String[] {DocumentConstants.Fields.DOC_ID});
+          ProjectionPath.from(DocumentConstants.Fields.DOC_ID));
     }
     if (add$vector) {
       buildPath(
           failOnOverlap,
           DocumentConstants.Fields.VECTOR_EMBEDDING_FIELD,
           root,
-          new String[] {DocumentConstants.Fields.VECTOR_EMBEDDING_FIELD});
+          ProjectionPath.from(DocumentConstants.Fields.VECTOR_EMBEDDING_FIELD));
     }
     if (add$vectorize) {
       buildPath(
           failOnOverlap,
           DocumentConstants.Fields.VECTOR_EMBEDDING_TEXT_FIELD,
           root,
-          new String[] {DocumentConstants.Fields.VECTOR_EMBEDDING_TEXT_FIELD});
+          ProjectionPath.from(DocumentConstants.Fields.VECTOR_EMBEDDING_TEXT_FIELD));
     }
     return root;
   }
 
   static void buildPath(
-      boolean failOnOverlap, String fullPath, ProjectionLayer layer, String[] segments) {
+      boolean failOnOverlap, String fullPath, ProjectionLayer layer, ProjectionPath path) {
+    List<String> segments = path.getSegments();
```

Comment:
Instead of accessor to `segments` (`List<String>`), should we instead have `String segment(int i)` accessor in `ProjectionPath` itself?